### PR TITLE
Initial Tenable Commit

### PR DESCRIPTION
**Pull request by richlilly2004**
Fixes #

## Proposed Changes
Tenable Playbooks, Parsers, Workbook and Detections from Tenable API
  -
  -
  -

### CODE REVIEW

**Review by richlilly2004 — COMMENTED** [review]
Updated with 30 day window for chart

File: Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+queryPeriod: 12h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+
+  Tenable_Assets_List
+  | join (Tenable_Vulns_List) 
+      on AssetID
+  | project Timestamp=TimeGenerated1, AssetID, FQDN, IP, Operating_system, VulnID, VulnName, plugin_family, SeverityType, Detection_source
+  | where SeverityType in ("Critical","High")
```
Comment (shainw) [inline Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml:20]: Generally you will want the where clause above the project for performance reasons.  I don't see any reason to NOT move it above the project here.

File: Detections/Tenable/Tenable - Low Vulnerability found.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+queryPeriod: 12h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+
+  Tenable_Assets_List
+  | join (Tenable_Vulns_List) 
+      on AssetID
+  | project Timestamp=TimeGenerated1, AssetID, FQDN, IP, Operating_system, VulnID, VulnName, plugin_family, SeverityType, Detection_source
+  | where SeverityType in ("Low")
```
Comment (shainw) [inline Detections/Tenable/Tenable - Low Vulnerability found.yaml:20]: Generally you will want the where clause above the project for performance reasons.  I don't see any reason to NOT move it above the project here.

File: Detections/Tenable/Tenable - Med Vulnerability found.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+queryPeriod: 12h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+
+  Tenable_Assets_List
+  | join (Tenable_Vulns_List) 
+      on AssetID
+  | project Timestamp=TimeGenerated1, AssetID, FQDN, IP, Operating_system, VulnID, VulnName, plugin_family, SeverityType, Detection_source
+  | where SeverityType in ("Medium")
```
Comment (shainw) [inline Detections/Tenable/Tenable - Med Vulnerability found.yaml:20]: Generally you will want the where clause above the project for performance reasons.  I don't see any reason to NOT move it above the project here.

**Review by shainw — CHANGES_REQUESTED** [review]
Some changes in the query.  Also, for validations to pass, you will need to implement a CustomTable JSON schema file that has the parser name and the schema resulting from the parser.  There are plenty of examples [here](https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables).  Documentation is [here](https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#datatypes)

**Comment by richlilly2004** [comment]
> Some changes in the query. Also, for validations to pass, you will need to implement a CustomTable JSON schema file that has the parser name and the schema resulting from the parser. There are plenty of examples [here](https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables). Documentation is [here](https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#datatypes)

Added

**Review by richlilly2004 — COMMENTED** [review]
Addressed open issues

**Review by sarah-yo — CHANGES_REQUESTED** [review]
Hi @richlilly2004, the some of the validations are still failing, please can you recheck? Thanks

**Comment by richlilly2004** [comment]
> Hi @richlilly2004, the some of the validations are still failing, please can you recheck? Thanks

Any guidance? Sorry, not sure quite how to read the issues on those.

File: Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+id: 6c0e1614-d41c-4c8f-b5f1-8b13e52c61da
+name: Tenable - Critical/High Vulnerability found
+description: |
+  'This query identifies a Tenable Vulnerability of Critical/High Vulnerability.'
+severity: High
+requiredDataConnectors:
+  - connectorId: Tenable
+    dataTypes: 
+      - Tenable_Assets_List
```
Comment (shainw) [inline Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml:9]: Based on your custom table, this needs to be Tenable_Assets_List_CL

File: Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+queryFrequency: 12h
+queryPeriod: 12h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+
+  Tenable_Assets_List
```
Comment (shainw) [inline Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml:16]: Based on your custom table, this needs to be Tenable_Assets_List_CL

File: Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  | join (Tenable_Vulns_List)
```
Comment (shainw) [inline Detections/Tenable/Tenable - CriticalHigh Vulnerability found.yaml:17]: Based on your custom table, this needs to be Tenable_Vulns_List_CL

File: Detections/Tenable/Tenable - Low Vulnerability found.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+id: 616aecbc-318f-4936-8267-bec55356e0bd
+name: Tenable - Low Vulnerability found
+description: |
+  'This query identifies a Tenable Vulnerability of Low Vulnerability.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: Tenable
+    dataTypes: 
+      - Tenable_Assets_List
```
Comment (shainw) [inline Detections/Tenable/Tenable - Low Vulnerability found.yaml:9]: Based on your custom table, this needs to be Tenable_Assets_List_CL

File: Detections/Tenable/Tenable - Low Vulnerability found.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+queryFrequency: 12h
+queryPeriod: 12h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+
+  Tenable_Assets_List
```
Comment (shainw) [inline Detections/Tenable/Tenable - Low Vulnerability found.yaml:16]: Based on your custom table, this needs to be Tenable_Assets_List_CL

File: Detections/Tenable/Tenable - Low Vulnerability found.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  | join (Tenable_Vulns_List)
```
Comment (shainw) [inline Detections/Tenable/Tenable - Low Vulnerability found.yaml:17]: Based on your custom table, this needs to be Tenable_Vulns_List_CL

File: Detections/Tenable/Tenable - Med Vulnerability found.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+id: 9d7b2b02-8b44-4850-89e9-a9ca442bc834
+name: Tenable - Medium Vulnerability found
+description: |
+  'This query identifies a Tenable Vulnerability of Medium Vulnerability.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Tenable
+    dataTypes: 
+      - Tenable_Assets_List
```
Comment (shainw) [inline Detections/Tenable/Tenable - Med Vulnerability found.yaml:9]: Based on your custom table, this needs to be Tenable_Assets_List_CL

File: Detections/Tenable/Tenable - Med Vulnerability found.yaml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+queryFrequency: 12h
+queryPeriod: 12h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+
+  Tenable_Assets_List
```
Comment (shainw) [inline Detections/Tenable/Tenable - Med Vulnerability found.yaml:16]: Based on your custom table, this needs to be Tenable_Assets_List_CL

File: Detections/Tenable/Tenable - Med Vulnerability found.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  | join (Tenable_Vulns_List)
```
Comment (shainw) [inline Detections/Tenable/Tenable - Med Vulnerability found.yaml:17]: Based on your custom table, this needs to be Tenable_Vulns_List_CL

**Review by shainw — CHANGES_REQUESTED** [review]

File: Detections/Tenable/Tenable - Med Vulnerability found.yaml (hunk 12 above)
Comment (richlilly2004) [reply Detections/Tenable/Tenable - Med Vulnerability found.yaml:17]: Hey @shainw I was assuming the parsers were imported, would you rather use _CL in the detections?

Comment (richlilly2004) [reply Detections/Tenable/Tenable - Med Vulnerability found.yaml:17]: Bump @shainw for thoughts on above

**Comment by shainw** [comment]
@preetikr - Since this requires a parser and a custom data connector, includes detections and a workbook, seems like this should be brought on as a Solution at this point, thoughts?

**Comment by preetikr** [comment]
@richlilly2004 - If you want to you can publish these as solutions. Here's the process of packaging and publishing a solution. https://github.com/Azure/Azure-Sentinel/tree/master/Solutions#guide-to-building-azure-sentinel-solutions

Thanks!

**Comment by richlilly2004** [comment]
Will do

**Comment by preetikr** [comment]
Thanks @richlilly2004 - please incorporate @shainw feedback as well so that the content can be merged. Also you'll want to move these commits to solutions folder to package and publish as solutions per guidance.

**Comment by shainw** [comment]
Very old PR, closing.

**Pull request closed without merging**